mpags-cipher.cpp

```diff
@@ -0,0 +1,244 @@
+    if (input_fileGiven) {
+
+        // At the moment, this will just be a print statement
+        std::cout << "The input file to the program is " << input_file << " ." << std::endl;
+        
+    }
+    else{
+        std::cout << "The input file must be specified. Use -h / --help for more information." << std::endl;
+         
+
+    }
+
+    // The output file
+    if (output_fileGiven) {
+
+        // At the moment, this will just be a print statement
+        std::cout << "The output file to the program is " << output_file << " ." << std::endl;
+        
+    }
+    else{
+
+        std::cout << "The output file must be specified. Use -h / --help for more information." << std::endl;
+         
+
+    }
+
+    // I'm keeping the user input part of my code for now.
+
+    // Initialising variables
+    char in_char{'x'};
+    std::string out_str{""};
+
+    // Take each letter from user input and in each case:
+    while(std::cin >> in_char)
+    {
+            
+        //- Convert to upper case
+        if(std::isalpha(in_char)){
+            out_str += std::toupper(in_char);
+            continue;
+        }
+
+        //- Change numbers to words
+        switch(in_char){
+
+            case '0':
+                out_str += "ZERO";
+                break;
+            
+            case '1':
+                out_str += "ONE";
+                break;
+            
+            case '2':
+                out_str += "TWO";
+                break;
+            
+            case '3':
+                out_str += "THREE";
+                break;
+
+            case '4':
+                out_str += "FOUR";
+                break;
+
+            case '5':
+                out_str += "FIVE";
+                break;
+
+            case '6':
+                out_str += "SIX";
+                break;
+
+            case '7':
+                out_str += "SEVEN";
+                break;
+
+            case '8':
+                out_str += "EIGHT";
+                break;
+
+            case '9':
+                out_str += "NINE";
+                break;
+            
+            default:
+                // Do nothing
+                break;
+        }
+
+        //- Ignore any other (non-alpha) characters
+        
+    }
+
+
+
+    // print out the new string
+    std::cout << out_str << std::endl;
+    
+}
+
+// I'm happy with submitting this as my attempt at the homework. 
+// The only problem that i know that still exists with my code is that, for instance, if typed
+// ./mpags-cipher -i -o my_output_file 
+// I know that my code would list -o as the input file and this is incorrect. 
+// I'm not sure how to get it to recognise that -o isn't a file.
+// I'm not sure i can do that at this stage because I'm not checking anything about the form of the argument that comes after -i or -o.
+//
+// I also kept the code for the user input from the terminal just because it worked and it doesn;t affect the other part of the code.
```

Comment:
Good, you were meant to leave that in, as part of the code for the cipher program that we're developing. The only code to discard from the first session was the trying out things with declaring variables etc. exercises.  From exercise 5 of the first session onwards we want to keep all the code we add (modulo some cases where it gets moved around or rewritten into a different form) since we're now building up the functionality of our cipher program.